### ISSUE TITLE
Improve format of region's bounding box

### ISSUE THREAD

**Issue by Carreau** [scikit-image/scikit-image#1959]
Just came across that with @stefanv while trying to detect a colorbar on a image. 
It would be easier when exploring writing code, instead of wondering which index is what.

**Comment by sdsingh** [comment]
Took a look at this, but I can't discern the attributes of bbox (which makes the original issue clear, haha).
The code states: 'Please refer to skimage.measure.regionprops` for more information,
but it is unclear to me where/what this is?

**Comment by jni** [comment]
https://github.com/scikit-image/scikit-image/blob/master/skimage/measure/_regionprops.py#L114

**Comment by stefanv** [comment]
We also realized that a named tuple is probably not feasible, given that
this code is N-d. Will instead update the description of bbox.

**Comment by jni** [comment]
What's the rationale for the ordering/shape of the output of bbox? I just had a closer look at this thanks to @sdsingh's PR and it seems to me to be a pretty useless format — not human readable *or* machine readable. I would say that `tuple((self._slice[i].start, self._slice[i].stop) for i in range(self._ndim)))` would be the right way to think about this...!

**Comment by sdsingh** [comment]
That also makes for sense to me, @jni, although I have little experience with this module.

**Comment by stefanv** [comment]
I agree.  Any suggestions on how to deprecate?

**Comment by jni** [comment]
@stefanv 

- add `bbox_new` method with new format
- add `bbox_old` method with existing format
- add deprecation warning `bbox` and return `bbox_old`, to be changed to `bbox_new` in 0.15. (This way people can start moving their code to use the new format.)

The function names are to be debated. for example, `bbox_new` could be instead `bounding_box`, which could remain an alias in perpetuity, and `bbox_old` could be `_bbox_old`, which could be deleted without affecting the public API.

**Comment by jni** [comment]
Incidentally, do we have a public method for returning the actual slices? I think many people would want to quickly get an object that they can use to index into a related array. So that's something else I would add.

**Comment by soupault** [comment]
Closed via #2030 .

**Comment by soupault** [comment]
:+1: to change the new output format for bbox.

@jni sounds like a reasonable idea, but that is really an "one-liner":
```
z = np.random.randn(25).reshape((5, 5))
z[[slice(a, b) for (a, b) in bbox_new]]
```

**Comment by JDWarner** [comment]
Just throwing this out there:

We can rather simply detect if a given input is one form vs. the other. Instead of a complicated deprecation over years for a bad API decision, we could (I would argue should)

* detect and handle both types of input, emitting a warning for the old type
* detect which type it is, but only handle the new version. Raise an informative error if the old bbox format is used, telling the user how to fix the problem.

Again - I'm usually all for API stability but complicated deprecations guaranteeing multiple code changes are worse than just fixing the problem. In this kind of instance where we can reliably detect which is which, I strongly favor just making the change.

**Comment by JDWarner** [comment]
I don't believe this issue was addressed in #2030; that just clarified current behavior.

**Comment by jni** [comment]
@JDWarner "detect the input"? It's an output that we are arguing about.

**Comment by stefanv** [comment]
@JDWarner Can you explain what you meant here?  Otherwise I think we should go with @jni's suggestion.
